### PR TITLE
Refactor - deleted syncSerial and syncConcurrent

### PR DESCRIPTION
**Pull request by yOoZZyOzzbourne**
After debating about concurrency modes, we agreed to these changes. Only mode left should be `asyncSerial`, main reason is that it does not block the `main queue` like the other two options.
So my refactor work deletes the two modes and leaves only `asyncSerial`.

**Before CR:** 
I left it like an enum, so that it doesnt break all the projects and if the project is already using asyncSerial (like it should), fix is not needed.

Readme + comments fixed.  

**EDIT:** 
 - Deleted `LoggingConcurrencyMode`, because its useful when we have only one mode.
 - Refactored `LoggerManager`
                  - deleted property `loggingConcurrencyMode`
                  - added property `serialQueue` from deleted code, so it works the same as it did before.
                  - changed `log` function, so it does what `asyncSerial` did. 
 - Fixed readme, changed `LoggingConcurrencyModes` to `Logging Execution`, left the `asyncSerial` here as an example.
 
Only changes that developers must do on a project is to **delete** the `loggingConcurrencyMode` from the inicialization (if its there).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -185,26 +185,14 @@ Log("This is the message to be logged.", onLevel: .info)
...
-#### `syncSerial`
-
-Logging task is dispatched synchronously on a custom serial queue, where all loggers perform their tasks serially one by one.
-
-![syncserial](https://user-images.githubusercontent.com/2511209/33495947-a2f21ca2-d6c8-11e7-9082-f841ef074012.png)
+`LoggerManager` supports now only one concurrency mode at the moment. By using `loggingConcurrencyMode` parameter when initializing `LoggerManager`, a user can set the `asyncSerial` mode, which is also a default value.
 
 #### `asyncSerial`
 
 Logging task is dispatched asynchronously on a custom serial queue, where all loggers perform their tasks serially one by one.
```
Comment (troupmar) [inline README.md]: Chtelo by se to bud uplne zbavit tech logging modes nebo aspon tyhle README casti a nebo napsat, ze zadny dalsi mody zatim nejsou podporovany. Takhle je zavadejici, ze je tu vycet modu ale ve vysledku je jen jeden. Chapu, zes to tak nechal kvuli zpetny kompatibilite ale mozna bych se s tim nesral a kompletne ty logging modes odstranil se vsim vsudy...

**Review by troupmar — CHANGES_REQUESTED** [review]

File: Sources/Logger/LoggerManager.swift [hunk 2]
```diff
@@ -73,9 +70,17 @@ public class LoggerManager {
...
+        
+        serialQueue.async {
+            availableLoggers.forEach { $0.log(log) }
+        }
     }
 }
+
+// MARK: - DispatchQueue + default queue
+
+public extension DispatchQueue {
```
Comment (troupmar) [inline Sources/Logger/LoggerManager.swift]: Premejslim, jestli musi bejt ta instance `public`, podle me by ji mel videt jen ten `LoggerManager`, takze `private` max `fileprivate`.

File: README.md [hunk 3]
```diff
@@ -183,28 +183,16 @@ Example of how to log:
...
+### Logging execution
 
-`LoggerManager` supports 3 different concurrency modes at the moment. By using `loggingConcurrencyMode` parameter when initializing `LoggerManager`, a user can set specific concurrency mode. Default value is `asyncSerial`.
-
-#### `syncSerial`
-
-Logging task is dispatched synchronously on a custom serial queue, where all loggers perform their tasks serially one by one.
-
-![syncserial](https://user-images.githubusercontent.com/2511209/33495947-a2f21ca2-d6c8-11e7-9082-f841ef074012.png)
+`LoggerManager` has its own style of executing tasks. It's done in form of `asyncSerial` style, that works on background thread without disrupting the main thread.
```
Comment (troupmar) [inline README.md]: Nemas cleny, `in the form of` a `works on the background thread`. Navic tam mas jen pulku pribehu, a to `async`. Napsal bych tam jeste vetu k ty druhy pulce pribehu, tj. `serial`. Logging into individual loggers is done in serial fashion, thus one by one. Nebo neco takovyho...

**Review by troupmar — CHANGES_REQUESTED** [review]

**Review by troupmar — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -183,28 +183,16 @@ Example of how to log:
...
-`LoggerManager` supports 3 different concurrency modes at the moment. By using `loggingConcurrencyMode` parameter when initializing `LoggerManager`, a user can set specific concurrency mode. Default value is `asyncSerial`.
-
-#### `syncSerial`
-
-Logging task is dispatched synchronously on a custom serial queue, where all loggers perform their tasks serially one by one.
-
-![syncserial](https://user-images.githubusercontent.com/2511209/33495947-a2f21ca2-d6c8-11e7-9082-f841ef074012.png)
+### Logging execution
 
+`LoggerManager` has its own style of executing tasks. It's done in the form of `asyncSerial` style, that works on the background thread without disrupting the main thread.
```
Comment (EllaGogo) [inline README.md]: Ja by som zmenila tuto sekci. `asyncSerial` uz v kodu realne neni, takze mi nedava smysl to spominat v readme. Muj navrh je tam mit jenom tohle:

### Logging execution
`LoggerManager` handles execution of logging tasks in an asynchronous serial manner. Each logging task is dispatched asynchronously on a custom serial background queue, where all loggers perform their tasks serially one by one.

**Review by EllaGogo — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (yOoZZyOzzbourne) [reply README.md]: chtěl jsem to tam nechat i s tím obrázkem ale asi to dává smysl to dát pryč

Comment (EllaGogo) [reply README.md]: obrazek tam muze zustat klidne, ale nechala by som ho v tej Logging execution sekci

**Review by EllaGogo — APPROVED** [review]

**Pull request merged**